The generated name is the same as the package name

<!-- Provide a general summary of the issue in the Title above -->
I created a table using `ent init Config`,
after i generate the code use `ent generate ./ent/schema`.
The program was compiled with error

error code:
```
ent\client.go:13:2: config redeclared in this block
	previous declaration at ent\config.go:14:6
```

<!-- Checked checkbox should look like this: [x] -->

- [x ] The issue is present in the latest release.
- [x ] I have searched the [issues](https://github.com/facebook/ent/issues) of this repository and believe that this is not a duplicate.

## Steps to Reproduce 🕹

Steps:

1. ent init Config
2. ent generate ./ent/schema
3. coding a main function
4. build and run it

## Your Environment 🌎

| Tech        | Version |
| ----------- | ------- |
| Go          | 1.15.6  |
| Ent         | 0.5.4  |
| Database    | sqlite3|
| Driver      | https://github.com/mattn/go-sqlite3        |

--- a/entc/gen/type.go
+++ b/entc/gen/type.go
@@ -694,6 +694,9 @@ func ValidSchemaName(name string) error {
 	if _, ok := globalIdent[name]; ok {
 		return fmt.Errorf("schema name conflicts with ent predeclared identifier %q", name)
 	}
+	if _, ok := privateField[pkg]; ok {
+		return fmt.Errorf("schema name conflicts with ent builder fields %q", pkg)
+	}
 	return nil
 }
 

